File: src/main/java/org/myfintech/payment/service/PaymentServiceFacade.java
package org.myfintech.payment.service;

import java.util.List;

import org.myfintech.payment.domain.PaymentCreateDTO;
import org.myfintech.payment.domain.PaymentDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface PaymentServiceFacade {

	void saveAsynch(String trackingNumber, List<PaymentDTO> dto);

	Page<PaymentDTO> findAll(Pageable pageable);

	PaymentDTO findById(Long id);

	PaymentDTO save(PaymentCreateDTO dto);

	PaymentDTO update(Long id, PaymentDTO dto);

	List<PaymentDTO> findPaymentsByContractNumber(String contractNumber);

	void validatePaymentsOrFail(List<PaymentDTO> payments);

}

File: src/main/kotlin/org/myfintech/payment/service/PaymentServiceFacade.kt
package org.myfintech.payment.service

import org.myfintech.payment.domain.PaymentCreateDTO
import org.myfintech.payment.domain.PaymentDTO
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface PaymentServiceFacade {
    fun saveAsynch(trackingNumber: String, dto: List<PaymentDTO>)

    fun findAll(pageable: Pageable): Page<PaymentDTO>

    fun findById(id: Long): PaymentDTO

    fun save(dto: PaymentCreateDTO): PaymentDTO

    fun update(id: Long, dto: PaymentDTO): PaymentDTO

    fun findPaymentsByContractNumber(contractNumber: String): List<PaymentDTO>

    fun validatePaymentsOrFail(payments: List<PaymentDTO>)
}
